test/server/cards/02_SHD/events/NoBargain.spec.ts

```diff
@@ -0,0 +1,35 @@
+describe('No Bargain', function() {
+    integration(function(contextRef) {
+        describe('No Bargain\'s ability', function() {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['no-bargain', 'wampa'],
+                    },
+                    player2: {
+                        hand: ['pillage', 'crafty-smuggler'],
+                    }
+                });
+            });
+
+            it('should draw a card and make opponent discard one', function () {
```

Comment:
If you could do me a favor - the discard system is new, we'd like to add a few extra tests on the first few cards that use it. Could you add one more case where the opponent has no cards in hand and confirm that the draw still happens?

Reply:
@AMMayberry1 Done!